core/00_primordials.js

```diff
@@ -34,6 +34,14 @@
 "use strict";
 
 (() => {
+  // Provide bootstrap namespace
+  globalThis.__bootstrap ??= {};
+  const key = Symbol.for("00_primordials.js");
+  if (globalThis.__bootstrap[key]) {
+    return;
+  }
+  globalThis.__bootstrap[key] = true;
```

Comment:
Do I get it right that this file can be actually executed multiple times when creating snapshots? Does it mean that all files that are snapshotted have this caveat or maybe it's only limited to "scripts" (ie. not ES modules).

Reply:
Yeah, it's just a trade-off of unifying `init_ops_and_esm()` and `init_ops()`. ES modules already have identities attached to their specifiers, with just the minor change of making `mod_evaluate()` idempotent.